### PR TITLE
SE-11 feature: PMM endpoint needs to be a single URL

### PR DESCRIPTION
**Pull request by pooknull**
https://jira.percona.com/browse/SE-11

Added option `pmm_url` and replaced `victoria_metrics_url`/`click_house_url` with bool options `victoria_metrics`/`clickhouse`

### CODE REVIEW

File: cmd/transferer/datasources.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	Password    string      `json:"password"`
+	User        string      `json:"user"`
+	Database    string      `json:"database"`
+	BasicAuth   bool        `json:"basicAuth"`
+	IsDefault   bool        `json:"isDefault"`
+	JsonData    interface{} `json:"jsonData"`
+	ReadOnly    bool        `json:"readOnly"`
+}
+
+type DataSources struct {
```
Comment (ollevche) [inline cmd/transferer/datasources.go:29]: rename to PMMConfig and fields

Comment (ollevche) [reply cmd/transferer/datasources.go:29]: it should contain:
PMM URL
CH URL
VM URL

**Review by ollevche — CHANGES_REQUESTED** [review]

File: cmd/transferer/datasources.go (hunk 1 above)
Comment (pooknull) [reply cmd/transferer/datasources.go]: Resolved in 8ddcf0da92e6d41ed9759a2e07e04f06f1581e4f

File: cmd/transferer/config.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	} else {
+		result.VictoriaMetricsURL = modifyURL(*pmmURL, "/prometheus", pmmURL.User, "", "")
+	}
+	if chLink != "" {
+		result.ClickHouseURL = chLink
+	} else {
+		result.ClickHouseURL = modifyURL(*pmmURL, "", nil, "9000", "database=pmm")
+	}
+
+	return result, nil
```
Comment (ollevche) [inline cmd/transferer/config.go:34]: Could be simplified, but test it first :)

```suggestion
	pmmURL, err := url.Parse(pmmLink)
	if err != nil {
		return result, fmt.Errorf("failed to parse pmm_url: %s", err)
	}
	
	var conf = PMMConfig{
		PmmURL: pmmLink,
		VictoriaMetricsURL: vmLink,
		ClickHouseURL: chLink,
	}

	if conf.VictoriaMetricsURL == "" {
		conf.VictoriaMetricsURL = modifyURL(*pmmURL, "/prometheus", pmmURL.User, "", "")
	}
	
	if conf.ClickHouseURL == "" {
		conf.ClickHouseURL = modifyURL(*pmmURL, "", nil, "9000", "database=pmm")
	}

	return conf, nil
```

**Review by ollevche — CHANGES_REQUESTED** [review]

**Pull request merged**